pageserver/src/tenant.rs

```diff
@@ -3003,6 +3029,32 @@ impl Tenant {
         Ok(tl)
     }
 
+    /// Helper for unit tests to branch a timeline with some pre-loaded states.
+    #[cfg(test)]
+    #[allow(clippy::too_many_arguments)]
+    pub async fn branch_timeline_test_with_states(
```

Comment:
Similarly `with_layers`.

This one could use more assertions about the Lsns, we must not populate Lsns below `start_lsn` because that situation is not supposed to happen (even with timeline detach ancestor that situation is never "seen" by pageserver).

Let's call `start_lsn` the `ancestor_lsn`, even if that is already overloaded naming with `ancestor_start_lsn` in the HTTP api.